### PR TITLE
Indoor map example update based on user feedback

### PR DESCRIPTION
**Pull request by hnativ13**

### CODE REVIEW

File: indoor-map/demo.html [hunk 1]
```diff
@@ -44,7 +44,7 @@ <h1>Show an Indoor Map</h1>
     </p>
 	
      <p>
-      More information on the use of the HERE Indoor Maps API is available <a href="https://developer.here.com/documentation/maps/dev_guide/topics/venues.html" target="_blank">here</a>.
+      More information on the use of the HERE Indoor Maps API is available <a href="https://developer.here.com/documentation/maps/3.1.34.2/dev_guide/topics/indoor-maps.html" target="_blank">here</a>.
```
Comment (dbacarel) [inline indoor-map/demo.html:47]:
```suggestion
      More information on the use of the HERE Indoor Maps API is available <a href="https://developer.here.com/documentation/maps/dev_guide/topics/indoor-maps.html" target="_blank">here</a>.
```

File: indoor-map/demo.js [hunk 2]
```diff
@@ -113,8 +109,5 @@ var behavior = new H.mapevents.Behavior(new H.mapevents.MapEvents(map));
 // Step 4: create the default UI component, for displaying bubbles
 var ui = H.ui.UI.createDefault(map, defaultLayers);
 
-// Indoor Maps provider interacts with a tile layer to visualize and control the Indoor Map
-const venuesProvider = new H.venues.Provider();
-
 // Step 5: add the Indoor Map
-addVenueToMap(map);
+addVenueToMap(map, platform);
```
Comment (dbacarel) [inline indoor-map/demo.js:113]: no need to pass the platform instance, it's already visible from inside the scope of `addVenueToMap`

```suggestion
addVenueToMap(map);
```

File: indoor-map/demo.js [hunk 3]
```diff
@@ -34,17 +34,13 @@ const labelTextPreferenceOverride = [
  *
  * @param  {H.Map} map A HERE Map instance
  */
-function addVenueToMap(map) {
+function addVenueToMap(map, platform) {
```
Comment (dbacarel) [inline indoor-map/demo.js:37]:
```suggestion
function addVenueToMap(map) {
```

File: indoor-map/demo.js [hunk 4]
```diff
@@ -34,17 +34,13 @@ const labelTextPreferenceOverride = [
...
+  // Indoor Maps provider interacts with a tile layer to visualize and control the Indoor Map
+  const venuesProvider = new H.venues.Provider();
+
   // Get an instance of the Indoor Maps service using a valid apikey for Indoor Maps
   const venuesService = platform.getVenuesService({ apikey: yourApikey, hrn: indoorMapHrn }, 2);
```
Comment (dbacarel) [inline indoor-map/demo.js:43]: why was the call to `getMapInfoList` removed?

File: indoor-map/demo.html [hunk 5]
```diff
@@ -66,4 +66,8 @@ <h3>Code</h3>
     </p>
     <script type="text/javascript" src='demo.js'></script>
   </body>
+
+  <p>
+    For complete descriptions of the parameteres shown in this example, see the <a href=https://developer.here.com/documentation/maps/3.1.37.1/api_reference/index.html target="_blank">API Reference</a>.
```
Comment (dbacarel) [inline indoor-map/demo.html:71]:
```suggestion
    For complete descriptions of the parameters shown in this example, see the <a href=https://developer.here.com/documentation/maps/dev_guide/api_reference/index.html target="_blank">API Reference</a>.
```

**Review by dbacarel — CHANGES_REQUESTED** [review]

File: indoor-map/demo.js (hunk 2 above)
Comment (hnativ13) [reply indoor-map/demo.js:113]: Interestingly, in the user feedback, which this ticket is based on, there's a remark saying: 

"Variable `platform` is a global variable used in function `addVenueToMap. It is declared below the function but this is easily missed. I was trying to copy the function to an existing app and so got confused why this was not working."

Passing the platform instance in this case is a recommendation I received from Andrii Heonia. 

Or is there still no need to pass the platform instance here?

File: indoor-map/demo.js (hunk 4 above)
Comment (hnativ13) [reply indoor-map/demo.js:43]: I'm not sure about this one. The change was part of a recommendation I received from Andrii Heonia.

File: indoor-map/demo.js (hunk 2 above)
Comment (dbacarel) [reply indoor-map/demo.js:113]: I discussed it with Andrii, this is fine

File: indoor-map/demo.js (hunk 4 above)
Comment (dbacarel) [reply indoor-map/demo.js:43]: Please revert this change

Comment (hnativ13) [reply indoor-map/demo.js]: Done

**Comment by dbacarel** [comment]
@hnativ13  the history is once again compromised + your previous changes about passing `platform` instance to `addVenueToMap` are gone

**Comment by hnativ13** [comment]
> @hnativ13 the history is once again compromised + your previous changes about passing `platform` instance to `addVenueToMap` are gone

Oh my, sorry, I thought this was the one you requested to revert. Shall I commit it again?

**Comment by dbacarel** [comment]
@hnativ13 , I created a separate branch with the expected changes https://github.com/heremaps/maps-api-for-javascript-examples/pull/148 and merged it. Closing this one

**Pull request closed without merging**